### DISCUSSION TITLE
Added Starcoder LLM

### DISCUSSION THREAD

**Discussion by gventuri** [sinaptik-ai/pandas-ai#60] [Announcements]
Yesterday, HuggingFace and bigcode jointly released Starcoder, a 3B parameter language model that has been fine-tuned using open source code. They claim that it is the best language model for generating code. We are pleased to announce that we have successfully implemented Starcoder in PandasAI!

Running it is as easy as this:
```
from pandasai.llm.starcoder import Starcoder

df = pd.DataFrame(your_dataframe)

llm = Starcoder(api_token="YOUR_HF_API_KEY")
pandas_ai = PandasAI(llm)
response = pandas_ai.run(df, "Your prompt goes here")
```

Give it a try and let us know how it feels!

**Comment by laynar** [comment]
Hi gventuri,

How to get API KEY. Where? Could you help to instruct one by one. Because, I am beginner. Thank!

**Comment by gventuri** [comment]
@laynar you need the HuggingFace APIs. Check it out here: https://huggingface.co/docs/huggingface_hub/guides/inference

**Reply by laynar** [reply]
Hi @gventuri 
Thank you so much.

**Comment by AzamBukhari** [comment]
Hello

Here is the code and I try to use it with the famous titanic daaset of seaborn library and it is throwing the following error. Can you advise Please:

Error:
TypeError: Cannot setitem on a Categorical with a new category (nan), set the categories first

Code:

# import libraries
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

#imorting pandas ai library
from pandasai import PandasAI
from pandasai.llm.starcoder import Starcoder

# Instatiate a LLM
from pandasai.llm.openai import OpenAI
llm = Starcoder(api_token="my API key removed due to privacy")

#here goes ny dataset
df = sns.load_dataset('titanic')

pandas_ai = PandasAI(llm)
response = pandas_ai.run(df, "Top ten highest fare")

**Comment by ngmisl** [comment]
I'm testing starcoder, and it does error out and gives no return

OpenAI works perfectly
